File: Objects/Worker.java
package Objects;

import java.util.UUID;
import java.util.Scanner;

public class Worker {

    private static int n = 0;
    private String ID = "";
    private String name = "";
    private String lastName = "";
    private String birthday = "";
    private String CURP = "";
    private String dateOfRegister = "";
    private String RFC = "";
    private int wage = 0;
    private String schedule = "";
    private String role = "";

    public Worker(String name, String lastName, String birthday, String CURP, String dateOfRegister, String RFC, int wage, String schedule, String role) {
        this.name = name;
        this.lastName = lastName;
        this.birthday = birthday;
        this.dateOfRegister = dateOfRegister;
        this.RFC = RFC;
        this.CURP = CURP;
        this.wage = wage;
        this.schedule = schedule;
        this.role = role;
        n++;
        setID();
    }

    public String getID(){
        return ID;
    }

    public void setID(){
        switch (this.role) {
            case "Technician" -> ID = "WT" + UUID.randomUUID().toString().substring(2, 12);
            case "Guide" -> ID = "WG" + UUID.randomUUID().toString().substring(2, 12);
            case "Administrator" -> ID = "WA" + UUID.randomUUID().toString().substring(2, 12);
            case "Veterinary" -> ID = "WV" + UUID.randomUUID().toString().substring(2, 12);
        }
    }

    public String getName() {
        return String.format("%s %S",name,lastName);
    }

    public void setName(String name, String lastName) {
        this.name = name;
        this.lastName = lastName;
    }


    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public String getCURP() {
        return CURP;
    }

    public void setCURP(String CURP) {
        this.CURP = CURP;
    }

    public String getDateOfRegister() {
        return dateOfRegister;
    }

    public void setDateOfRegister(String dateOfRegister) {
        this.dateOfRegister = dateOfRegister;
    }

    public String getRFC() {
        return RFC;
    }

    public void setRFC(String RFC) {
        this.RFC = RFC;
    }

    public int getWage() {
        return wage;
    }

    public void setWage(int wage) {
        this.wage = wage;
    }

    public String getSchedule() {
        return schedule;
    }

    public void setSchedule(String schedule) {
        this.schedule = schedule;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
        setID();
    }
}

File: Worker.py
import uuid

class Worker:
    __id = " "
    __name = " "
    __last_name = " "
    __birthday = " "
    __curp = " "
    __date_of_register = " "
    __rfc = " "
    __wage = 0
    __schedule = " "
    __role = " "

    def __init__(self, name, last_name, birthday, curp, date_of_register, rfc, wage, schedule, role):
        self.__name = name
        self.__last_name = last_name
        self.__birthday = birthday
        self.__curp = curp
        self.__date_of_register = date_of_register
        self.__rfc = rfc
        self.__wage = wage
        self.__schedule = schedule
        self.__role = role
        self.__id = self.set_id()

    def set_id(self):
        if self.__role == "Technician" :
            xd = uuid.uuid4()
            id = "WT" + str(xd)
            id = id[0:12]
        elif self.__role == "Guide" :
            xd = uuid.uuid4()
            id = "WG" + str(xd)
            id = id[0:12]
        elif self.__role == "Administrator" :
            xd = uuid.uuid4()
            id = "WA" + str(xd)
            id = id[0:12]
        elif self.__role == "Veterinary" :
            xd = uuid.uuid4()
            id = "WV" + str(xd)
            id = id[0:12]
        return id

    def set_name(self, name, last_name):
        self.__name = name
        self.__last_name = last_name

    def set_birthday(self, birthday):
        self.__birthday = birthday
        self.__id = self.set_id()

    def set_curp(self, curp):
        self.__curp = curp

    def set_date_of_register(self, date_of_register):
        self.__date_of_register = date_of_register

    def get_id(self):
        return id

    def get_name(self):
        return self.__name + " " + self.__last_name
    
    def get_birthday(self):
        return self.__birthday
    
    def get_curp(self):
        return self.__curp
    
    def get_date_of_register(self):
        return self.__date_of_register
    
    def get_rfc(self):
        return self.__rfc
    
    def set_rfc(self, rfc):
        self.__rfc = rfc

    def get_wage(self):
        return self.__wage
    
    def set_wage(self, wage):
        self.__wage = wage

    def get_schedule(self):
        return self.__schedule

    def set_schedule(self, schedule):
        self.__schedule = schedule

    def get_role(self):
        return self.__role
    
    def set_role(self, role):
        self.__role = role
        self.__id = self.set_id()

    def show_slave(self):
        print("ID: "+ self.__id +   "  Full name: " + self.__name + " " + self.__last_name +  "  Birthday: " + self.__birthday +  
        "  Date of register: " + self.__date_of_register + "  CURP: " + self.__curp +  "  RFC: "+ self.__rfc +
        "  Role: " + self.__role + "  Wage: " + str(self.__wage) + "  Schedule:  " + self.__schedule)

